### PR TITLE
Scroll: OZ audit

### PR DESCRIPTION
**Pull request by isabellewei**
Contract code updates for the OZ audits:
- added docstrings to `ScrollBridgeReceiver` and `IScrollMessenger`
- emit an event when constructing `ScrollBridgeReceiver`

Other fly-by changes:
- update blockscout URL
- update roots from running migration

### CODE REVIEW

File: contracts/bridges/scroll/ScrollBridgeReceiver.sol [hunk 1]
```diff
@@ -9,19 +9,26 @@ contract ScrollBridgeReceiver is SweepableBridgeReceiver {
 
     event Newl2Messenger(address indexed oldL2Messenger, address indexed newL2Messenger);
```
Comment (kevincheng96) [inline contracts/bridges/scroll/ScrollBridgeReceiver.sol:10]: nit: L2 should be capitalized

File: contracts/bridges/scroll/ScrollBridgeReceiver.sol [hunk 2]
```diff
@@ -9,19 +9,26 @@ contract ScrollBridgeReceiver is SweepableBridgeReceiver {
...
+    /// @notice Construct a new ScrollBridgeReceiver instance
+    /// @param l2Messenger_ Address of Scroll L2 Messenger contract
     constructor(address l2Messenger_) {
         l2Messenger = l2Messenger_;
+        emit Newl2Messenger(address(0), l2Messenger_);
     }
 
+    /// @notice Update the L2 Messenger address
+    /// @param newL2Messenger New address for L2 Messenger contract
     function changel2Messenger(address newL2Messenger) public {
```
Comment (kevincheng96) [inline contracts/bridges/scroll/ScrollBridgeReceiver.sol:24]: nit: L2 should be capitalized

File: contracts/bridges/scroll/ScrollBridgeReceiver.sol
Comment (kevincheng96) [inline contracts/bridges/scroll/ScrollBridgeReceiver.sol:1]: Missing contract level natspec

File: contracts/Comet.sol [hunk 3]
```diff
@@ -1121,7 +1121,6 @@ contract Comet is CometMainInterface {
         LiquidatorPoints memory points = liquidatorPoints[absorber];
         points.numAbsorbs++;
         points.numAbsorbed += safe64(accounts.length);
-        points.approxSpend += safe128(gasUsed * block.basefee);
```
Comment (kevincheng96) [inline contracts/Comet.sol:1124]: Curious if you are planning on addressing L-03 as well?

Comment (isabellewei) [reply contracts/Comet.sol:1124]: I'm not sure why this diff is included here actually, the line has already been deleted in this repo.

I decided not to address L-03, I think it's better to make as few changes to the original Comet contract as possible

File: contracts/bridges/scroll/ScrollBridgeReceiver.sol [hunk 4]
```diff
@@ -4,10 +4,12 @@ pragma solidity 0.8.15;
 import "../SweepableBridgeReceiver.sol";
 import "./IScrollMessenger.sol";
 
+/// @title Scroll Bridge Receiver
+/// @notice Contract to facilitate message passing from the Comet protocol to Scroll's bridge
```
Comment (kevincheng96) [inline contracts/bridges/scroll/ScrollBridgeReceiver.sol:8]: nit: "Contract that processes messages passed from Compound governance using the Scroll bridge"

**Review by kevincheng96 — APPROVED** [review]
One more suggestion to change a natspec. Otherwise, LGTM!

**Pull request merged**